src/util/string-utils.c

```diff
@@ -81,7 +81,7 @@ parse_string_into_data(uint8_t* buffer, size_t len, const char* c_str)
 		len = 0;
 	}
 
-	while (*c_str != 0) {
+	while ((*c_str != 0) && (len > 0)) {
```

Comment:
Do we need to verify `c_str` is null-terminated and even-length?

Reply:
Yes. `len` is for the size of `buffer`. `c_str` is a zero-terminated c-string, and has no explicit length. We must check both at each step.